include/envoy/config/typed_metadata.h

```diff
@@ -74,6 +74,7 @@ class TypedMetadataFactory {
   parse(const ProtobufWkt::Struct& data) const PURE;
 
   static std::string category() { return "typed_metadata"; }
+  static std::string type() { return ""; }
```

Comment:
qq from a skim: what does `type()` mean and why is it set to empty string everywhere?

Reply:
It's set to "" for factories that don't have protos defined.
type() means the protobuf message name for the config of a factory.

Reply:
Is there some way to make this more obvious and less boilerplate either via templating, base class with a default, etc.?

Reply:
the happy case is handled by TypedConfig base class. I was thinking we may want to revisit the unhappy cases and fix them, so wasn't optimizing for that. we can provide a default base class UntypedConfig for that.

Reply:
Yeah something as simple as UntypedConfig IMO would make it a lot more clear, but up to you.

Reply:
Would it make sense to have a common base class for all factories at this point? I think there might be a critical mass of attributes common to all factories.

Reply:
Yeah, category, name, and type are sufficient for a base class.